gammapy/scripts/analysis.py

```diff
@@ -0,0 +1,76 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Session class driving the high-level interface API"""
+import jsonschema
+import logging
+from ..utils.scripts import read_yaml
+from astropy import units as u
+from pathlib import Path
+
+__all__ = ["Analysis"]
+
+log = logging.getLogger(__name__)
+CONFIG_PATH = Path(__file__).resolve().parent / "config"
+CONFIG_FILE = CONFIG_PATH / "default.yaml"
+SCHEMA_FILE = CONFIG_PATH / "schema.yaml"
+
+
+class Analysis():
+    """High-level interface analysis session class.
+
+    The Analysis class drives an analysis working session using the high-level interface
+    API. It is initialized by default with a set of configuration parameters and values
+    declared in an internal configuration YAML file.
+
+    Parameters
+    ----------
+    configfile : string
+        The name of a user defined configuration file.
+    """
+
+    def __init__(self, configfile=CONFIG_FILE, **kwargs):
+        self.config = dict()
+        if configfile:
+            self.configfile = configfile
+            self.config = read_yaml(self.configfile)
+        if len(kwargs):
+            self.config.update(**kwargs)
+        self._set_logging()
+
+    def validate_config(self):
+        """Validate config parameters against schema."""
+        schema = read_yaml(SCHEMA_FILE)
+        try:
+            jsonschema.validate(self.config, schema, _gp_validator)
+        except jsonschema.exceptions.ValidationError as ex:
+            log.error("Error when validating configuration parameters against schema.")
+            log.error("Parameter: {}".format(ex.schema_path[-2]))
+            log.error(ex.message)
+            raise ex
+
+    def _set_logging(self):
+        """Set logging parameters for API."""
+        if "global" in self.config and "logging" in self.config["global"]:
```

Comment:
We should get rid of the `if` here. If we start to write code like this everywhere we get a config value, we're in hell. I'll make a suggestion how to handle this below.